assertj-core/src/test/java/org/assertj/core/internal/classes/Classes_assertIsNotPrimitive_Test.java

```diff
@@ -0,0 +1,118 @@
+package org.assertj.core.internal.classes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.error.ShouldBePrimitive.shouldNotBePrimitive;
+import static org.assertj.core.test.TestData.someInfo;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.ClassesBaseTest;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for
+ * <code>{@link org.assertj.core.internal.Classes#assertIsNotPrimitive(AssertionInfo, Class)}</code> .
+ *
+ * @author Manuel Gutierrez
+ */
+class Classes_assertIsNotPrimitive_Test extends ClassesBaseTest {
+
+  @Test
+  void should_pass_if_actual_is_an_object() {
+    // GIVEN
+    actual = Object.class;
+    // WHEN/THEN
+    classes.assertIsNotPrimitive(someInfo(), actual);
+  }
+
+  @Test
+  void should_fail_if_actual_is_null() {
+    // GIVEN
+    actual = null;
+    // WHEN
+    AssertionError assertionError = expectAssertionError(() -> assertThat(actual).isNotPrimitive());
+    // THEN
+    then(assertionError).hasMessage(actualIsNull());
+  }
+
+  @Test
+  void should_fail_if_actual_has_byte_type() {
```

Comment:
there is nothing wrong with these tests but I think it's a good fit for a parameterized test taking the list of primitive type as input, this should work: 
```java
  @ParameterizedTest
  @ValueSource(classes = { int.class, double.class }) // list all primitive classes
  void should_fail_if_actual_is_a_primitive_type(Class<?> primitiveClass) {
   //WHEN
    AssertionError assertionError = expectAssertionError(() -> assertThat(primitiveClass).isNotPrimitive());
    //THEN
    then(assertionError).hasMessage(shouldNotBePrimitive(primitiveClass).create());
  }
```
Once this is done, you can remove the other `should_fail_if_actual_has_...` tests